ipv8/REST/dht_endpoint.py

```diff
@@ -20,10 +25,126 @@ def __init__(self, session):
         resource.Resource.__init__(self)
 
         dht_overlays = [overlay for overlay in session.overlays if isinstance(overlay, DHTCommunity)]
+        tc_overlays = [overlay for overlay in session.overlays if isinstance(overlay, TrustChainCommunity)]
         if dht_overlays:
-            self.putChild("statistics", DHTStatisticsEndpoint(dht_overlays[0]))
-            self.putChild("values", DHTValuesEndpoint(dht_overlays[0]))
-            self.putChild("peers", DHTPeersEndpoint(dht_overlays[0]))
+            self.putChild(b"statistics", DHTStatisticsEndpoint(dht_overlays[0]))
+            self.putChild(b"values", DHTValuesEndpoint(dht_overlays[0]))
+            self.putChild(b"peers", DHTPeersEndpoint(dht_overlays[0]))
+            self.putChild(b"block", DHTBlockEndpoint(dht_overlays[0], tc_overlays[0]))
+
+
+class DHTBlockEndpoint(resource.Resource, BlockListener):
+    """
+    This endpoint is responsible for returning the latest Trustchain block of a peer. Additionally, it ensures
+    this peer's latest TC block is available
+    """
+
+    def received_block(self, block):
+        """
+        Wrapper callback method, inherited from the BlockListener abstract class, which will publish the latest
+        TrustChain block to the DHT
+
+        :param block: the latest block added to the Database. This is not actually used by the inner method
+        :return: None
+        """
+        self.publish_latest_block()
+
+    def should_sign(self, block):
+        pass
+
+    KEY_SUFFIX = b'_BLOCK'
+
+    def __init__(self, dht, trustchain):
+        resource.Resource.__init__(self)
+        self.dht = dht
+        self.trustchain = trustchain
+        self.block_version = 0
+
+        self._hashed_dht_key = sha1(self.trustchain.my_peer.mid + self.KEY_SUFFIX).digest()
+
+        trustchain.add_listener(self, [trustchain.UNIVERSAL_BLOCK_LISTENER])
+
+    def reconstruct_all_blocks(self, block_chunks):
+        """
+        Given a list of block chunks, reconstruct all the blocks in a dictionary indexed by their version
+
+        :param block_chunks: the list of block chunks
+        :return: a dictionary of reconstructed blocks (in packed format), indexed by the version of the blocks,
+                 and the maximal version
+        """
+        new_blocks = {}
+        max_version = 0
+
+        for entry in block_chunks:
+            this_version = struct.unpack("I", entry[1:3] + b'\x00\x00')[0]
```

Comment:
Why not just make this an unsigned short ('H') so you don't have to append '\x00\x00'?